x/launch/keeper/grpc_query_genesis_account.go

```diff
@@ -28,7 +28,9 @@ func (k Keeper) GenesisAccountAll(c context.Context, req *types.QueryAllGenesisA
 			return err
 		}
 
-		genesisAccounts = append(genesisAccounts, &genesisAccount)
+		if req.ChainID == "" || genesisAccount.ChainID == req.ChainID {
```

Comment:
For performance reason we should use `query.Paginate` with the longest common prefix. So in this case the prefix that include the `chainID`

For this we can add a method that returns the prefix with `chainID` in `types/key_[accounttype].go` then we provide this prefix to the `prefix.NewStore` method above